src/transform/globalize.js

```diff
@@ -29,30 +28,75 @@ function indent (code) {
   return lines.join('\n');
 }
 
-function relative (file) {
+function makePathRelative (file) {
   return path.relative(process.cwd(), file);
 }
 
+function defineDependencies (imports, dependencies) {
+  var code = '';
+  var keyVals = imports.map(function (imp, idx) {
+    return '"' + imp + '": ' + generateModuleName(dependencies[idx]);
+  });
+
+  keyVals.unshift('"exports": exports');
+  keyVals.unshift('"module": module');
+  code = '\n' + indent(keyVals.join(',\n')) + '\n';
+  return '{' + code + '}';
+}
+
+function defineReplacement (name, deps, func) {
+  var rval;
+  func = [func, deps, name].filter(function (cur) { return typeof cur === 'function'; })[0];
+  deps = [deps, name, []].filter(Array.isArray)[0];
+  rval = func.apply(null, deps.map(function (value) {
+    return defineDependencies[value];
+  }));
+  return rval && (exports = module.exports = rval);
```

Comment:
When using Babel, it doesn't check it's type and immediately tries to call `Object.defineProperty` on it. Perhaps we should be a bit more stringent here.